tts-evaluation-MCD-DTW.ipynb

```diff
@@ -0,0 +1,495 @@
+    "import librosa\n",
+    "import librosa.display\n",
+    "import numpy as np\n",
+    "import math\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "398b01e0",
+   "metadata": {},
+   "source": [
+    "### Define parameters for spectrogram generation.\n",
+    "In this section we define parameters required to generate our spectrograms. More information about these parameteres can found on librosa documentation for [stft](https://librosa.org/doc/main/generated/librosa.stft.html), [mels](https://librosa.org/doc/main/generated/librosa.filters.mel.html) and [mfcc](https://librosa.org/doc/main/generated/librosa.feature.mfcc.html) generation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "191012de",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Mel params\n",
+    "n_fft=1024\n",
+    "hop_length=256\n",
+    "win_length=None\n",
+    "window='hann'\n",
+    "n_mels = 80\n",
+    "sr = 22050\n",
+    "\n",
+    "## Mfcc params\n",
+    "n_mfcc=n_mels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54830c8c",
+   "metadata": {},
+   "source": [
+    "A little bit about these parameters:  \n",
+    "    - `n_fft`: Number of fft_components or length of windowed signal after padding in stft.  \n",
+    "    - `hop_length`: Number of audio samples between adjacent STFT columns.  \n",
+    "    - `window`: Window to use in stft.  \n",
+    "    - `win_length`: Length of the window to be used.  \n",
+    "    - `n_mels`: Number of number of Mel bands to generate.  \n",
+    "    - `sr`: Sample rate of the samples.  \n",
+    "    - `n_mfcc`: Number of MFCCs to generate."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4cb9e37c",
+   "metadata": {},
+   "source": [
+    "### Load and Visualize data\n",
+    "Lets apply the algorithm on a synthesized, ground truth mels pair for understanding."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8546edb5",
+   "metadata": {},
+   "source": [
+    "Write a function to convert mels to mfcc, we will use [mfcc](https://librosa.org/doc/main/generated/librosa.feature.mfcc.html) generation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c1a0663d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def mel2mfcc(mels):\n",
+    "    mfcc = librosa.feature.mfcc(S=mels, n_mfcc=n_mfcc)\n",
+    "    mfcc = librosa.power_to_db(mfcc, ref=np.max)\n",
+    "    return mfcc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6efac554",
+   "metadata": {},
+   "source": [
+    "For this tutorial, we have already generated mels for a [fastpitch](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_en_fastpitch) model and [radtts](https://github.com/NVIDIA/radtts) model, we have also generated mels for ground truth audio using librosa [mel extractor](https://librosa.org/doc/main/generated/librosa.filters.mel.html)"
```

Comment:
Can you generate the mels from the ground audio, rather than using pre-computed ones?